core/src/main/java/rdx/works/core/preferences/PreferencesManager.kt

```diff
@@ -167,7 +167,7 @@ class PreferencesManagerImpl @Inject constructor(
                 FactorSourceId.Hash(
                     FactorSourceIdFromHash(kind = FactorSourceKind.DEVICE, body = Exactly32Bytes.init(hex.hexToBagOfBytes()))
                 )
-            }.toSet()
+            }.ifEmpty { emptySet() }.toSet()
```

Comment:
Can the list be empty at this point? It doesn't seem so from the `.orEmpty` method call above.